src/journal-remote/journal-upload.c

```diff
@@ -180,29 +183,23 @@ int start_upload(Uploader *u,
         assert(input_callback);
 
         if (!u->header) {
-                struct curl_slist *h;
+                _cleanup_(curl_slist_free_allp) struct curl_slist *h = NULL;
 
                 h = curl_slist_append(NULL, "Content-Type: application/vnd.fdo.journal");
                 if (!h)
                         return log_oom();
 
-                h = curl_slist_append(h, "Transfer-Encoding: chunked");
-                if (!h) {
-                        curl_slist_free_all(h);
+                if (!curl_slist_append(h, "Transfer-Encoding: chunked"))
```

Comment:
Unless I'm reading the documentation wrong, this needs to save the return value to a temp variable, which then needs to overwrite `h` ? (what a convoluted api...)

https://curl.se/libcurl/c/curl_slist_append.html

Reply:
Well, it's hard to imagine that it could return anything but the same list or NULL (and it doesn't ATM)... But I can change the code, of course.

Reply:
Yes, it would be strange - but it's an explicit suggestion in the API doc, so I figure better to be safe and follow it.

Reply:
Updated.